extensions/amp-subscriptions/0.1/viewer-subscription-platform.js

```diff
@@ -85,11 +85,11 @@ export class ViewerSubscriptionPlatform {
       'productId': this.currentProductId_,
       'origin': this.origin_,
     });
-    // TODO(chenshay): Viewer Matching: We don't know which viewer it actually
-    // is. Need to check the viewerUrl to know, or more specificlly iterate via
-    // configured platforms and check whether any of these support the viewer.
+    const viewerUrl = this.viewer_.getResolvedViewerUrl().includes('google.com')
```

Comment:
I don't know, unfortunately, if this is currently guaranteed to exist. I think a better chance to do this is using `viewer.maybeGetMessagingOrigin`. But I'm not sure at this point what's the best way to confirm this. We may need to debug GNews.

Reply:
Modified the code to check for a "cryptokeys" param in the URL and then look in the cryptokeys JSON object for the keys inside of the param. Currently, we fall back to google.com but I plan on removing that in the future.